shared/is-docker.sh

```diff
@@ -20,6 +20,7 @@ should_run="$(${HERE}/scripts/shared/resinci-read.sh \
 if [[ -f "repo.yml" ]]; then
   project_type="$(yq read repo.yml 'type')"
   [[ "$project_type" == "generic" ]] && exit 1
+  [[ "$project_type" == "docker-compose" ]] && exit 1
```

Comment:
Since this is growing, I would suggest to move all this checks to a function that returns a single value. In that case, you have a single exit point which improves readability (but in that case you'll need to disable `-e` for that function...)

Reply:
@kaisoz this is currently a bit of a mess. We need to unify everything into detectorist and use that consistently for all project type detections. We currently have little bits all over the place that are duplicating the project type detection logic